data-plane/core/service/src/session/point_to_point.rs

```diff
@@ -656,29 +772,280 @@ where
+                    session_id,
+                    msg_rtx_id,
+                    &Some(self.state.source.clone()),
+                    &Some(self.state.unicast_name.as_ref().unwrap_or(&pkt_dst).clone()),
+                ));
+
+                Message::new_publish_with_headers(
+                    slim_header,
+                    session_header,
+                    "",
+                    payload.blob.to_vec(),
+                )
+            }
+            None => {
+                // the packet does not exist return an empty RtxReply with the error flag set
+                debug!(
+                    "received an RTX messages for an old packet on session {}",
+                    session_id
+                );
+
+                let flags = SlimHeaderFlags::default()
+                    .with_forward_to(incoming_conn)
+                    .with_error(true);
+
+                let slim_header = Some(SlimHeader::new(&pkt_dst, &pkt_src, Some(flags)));
+
+                // no need to set source and destiona here
+                let session_header = Some(SessionHeader::new(
+                    ProtoSessionType::SessionMulticast.into(),
+                    ProtoSessionMessageType::RtxReply.into(),
+                    session_id,
+                    msg_rtx_id,
+                    &None,
+                    &None,
+                ));
+
+                Message::new_publish_with_headers(slim_header, session_header, "", vec![])
+            }
+        };
+
+        self.state.tx.send_to_slim(Ok(rtx_pub)).await
+    }
+
+    async fn process_incoming_rtx_reply(&mut self, message: Message) -> Result<(), SessionError> {
+        // Remove RTX timer
+        let msg_id = message.get_session_header().get_message_id();
+        self.stop_and_remove_timer(msg_id, false)?;
+
+        let ack = self.create_ack(&message);
+
+        // Forward the message to the app
+        self.send_message_to_app(message).await?;
+
+        // Send the ack
         self.state
             .tx
-            .send_to_app(Ok(message))
+            .send_to_slim(Ok(ack))
             .await
             .map_err(|e| SessionError::SlimTransmission(e.to_string()))
     }
 
+    fn create_ack(&self, message: &Message) -> Message {
+        let src = message.get_source();
+        let message_id = message.get_session_header().message_id;
+        let slim_header = Some(SlimHeader::new(
+            &self.state.source,
+            &src,
+            Some(SlimHeaderFlags::default().with_forward_to(message.get_incoming_conn())),
+        ));
+
+        let session_header = Some(SessionHeader::new(
+            ProtoSessionType::SessionPointToPoint.into(),
+            ProtoSessionMessageType::P2PAck.into(),
+            message.get_session_header().session_id,
+            message_id,
+            &None,
+            &None,
+        ));
+
+        Message::new_publish_with_headers(slim_header, session_header, "", vec![])
+    }
```

Comment:
NOt for this PR, but we should probably consolidate all the creation of these control packet in one place and reuse actoss sessions.